src/libraries/System.IO.FileSystem/tests/ManualTests/ManualTests.cs

```diff
@@ -0,0 +1,75 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using Microsoft.DotNet.XUnitExtensions;
+using Xunit;
+
+namespace System.IO.ManualTests
+{
+    public class FileSystemManualTests
+    {
+        public static bool ManualTestsEnabled => !string.IsNullOrEmpty(Environment.GetEnvironmentVariable("MANUAL_TESTS"));
+        
+        [ConditionalFact(nameof(ManualTestsEnabled))]
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        public static void Throw_FileStreamDispose_WhenRemoteMountRunsOutOfSpace()
+        {
+            /*
+            Pre-requisites to run this test:
+
+            - The remote machine has a folder ~/share, where a small 1MB drive is mounted.
+            - The remote drive is almost full.
+            - The remote drive has one file, called "copyme.txt", large enough that if we attempt to programatically copy it into
+              the same folder, the action should fail because the file is slightly larger than the available free space in the disk.
+            - The local machine has a folder named "mountedremote" located in the local user's home folder.
+            - The remote folder is mounted into "mountedremote".too
+
+
+            Example of mounting a remote folder using sshfs and two Linux machines:
+
+            In remote machine:
+                - Install openssh-server.
+                - Create a partition of 1 MB. You can use gparted and a spare USB drive. The filesystem does not seem to matter, but the bug could be repro'd with ext4.
+                - Mount the drive in the 
+                - Fill the new drive with files until almost full, leave a couple of bytes free.
```

Comment:
How much of this can be automated? For example, filling the remote drive with files seems very automatable.

Can the pre-requities for the test be simplified to "1MB remote mounted drive with ext4 file-system" ?

Reply:
Sure. I automated filling the drive with the necessary files with the exact required length (assuming the drive was correctly created with a 1MB size).
I updated the instructions as requested too.